UefiTestingPkg/FunctionalSystemTests/MemoryProtectionTest/App/MemoryProtectionTestApp.c

```diff
@@ -897,59 +940,57 @@ UefiPageGuard (
     }
 
     UT_ASSERT_NOT_EFI_ERROR (mNonstopModeProtocol->ResetPageAttributes ());
+  } else {

```

Comment:
Since `MemoryProtectionContext.TestingMethod` is an enum which is an integer, it would help more strictly enforce correctness of the value to explicitly check for `MemoryProtectionTestWarmReset` here and then assert if any other value is present.

Reply:
UT_LOG_ERROR ("Invalid testing method specified: %d\n", MemoryProtectionContext.TestingMethod);
return UNIT_TEST_ERROR_TEST_FAILED;

added if the test method is unexpected. Explicit check for MemoryProtectionTestReset added